shell/browser/ui/file_dialog_win.cc

```diff
@@ -144,10 +144,14 @@ static void ApplySettings(IFileDialog* dialog, const DialogSettings& settings) {
   // We set file extension to the first none-wildcard extension to make
   // sure the dialog will update file extension automatically.
   for (size_t i = 0; i < filterspec.size(); ++i) {
-    if (std::wstring(filterspec[i].pszSpec) != L"*.*") {
+    std::wstring spec(filterspec[i].pszSpec);
+    if (spec != L"*.*") {
       // SetFileTypeIndex is regarded as one-based index.
       dialog->SetFileTypeIndex(i + 1);
-      dialog->SetDefaultExtension(filterspec[i].pszSpec);
+      // "*.jpg;*.png" => "*.jpg"
+      std::wstring first_spec = spec.substr(0, spec.find(L';'));
```

Comment:
does this still work when there's only one extension? as in "*.jpg" with no ';'

Reply:
> does this still work when there's only one extension? as in "*.jpg" with no ';'

Yes, this code will still work when there’s only one extension (e.g., `"*.jpg"` without a `';'`). The `spec.find(L';')` call will return `std::wstring::npos` in that case, so `substr(0, std::wstring::npos)` will simply return the entire string, `"*.jpg"`. Thus, it handles both cases correctly.